### ISSUE TITLE
Expose AbstractExpression as an usable class

### ISSUE THREAD

**Issue by SkytAsul** [Osmerion/Quitte#4]
Currently, AbstractExpression is a sealed class (its constructor is package-private).
I'd like to use it to make an observable which yields no value but notifies invalidation listeners whenever dependencies are updated. My plan was to extends AbstractExpression and add the correct dependencies and voilà, but as the constructor is not accessible my only possibility is to copy and paste pretty much all the content of AbstractExpression to a new class.
Is there a better way?

**Comment by TheMrMilchmann** [comment]
I might need more information about your use case, but initially, I think you should just register a single `InvalidationListener` to multiple observables. This works roughly as follows:

```java
ObservableIntValue a = ...;
ObservableIntValue b = ...;
        
InvalidationListener listener = observable -> {
    // logic ... 
};
        
a.addInvalidationListener(listener);
b.addInvalidationListener(listener);
```

Depending on your requirements, you might want to wrap the listener in a `WeakInvalidationListener`.

**Comment by SkytAsul** [comment]
That wouldn't work, I'd like to create a whole new Observable.

```java
ObservableIntValue a = ...;
ObservableIntValue b = ...;

Observable multi = /* that is the question */;
```
The goal is that all invalidation listeners of the "multi" Observable get notified whenever a _or_ b are updated.

**Comment by TheMrMilchmann** [comment]
This is implemented in fbf2f546fdd689bf388e035612841f8be76c7b42 and will land in 0.6.0 together with some additional cleanup for expressions. I didn't open up the API fully yet in order to avoid breaking changes in the future but everything needed should be there.
